Kernel/Arch/init.cpp

```diff
@@ -294,13 +335,15 @@ void init_stage2(void*)
 
     WorkQueue::initialize();
 
+#if ARCH(X86_64)
```

Comment:
I think this is fine for now but I'd like to see these types of arch-specific checks moved into generic-sounding functions we can implement in arch-specific files rather than have #ifdefs littering the code.

e.g.
```
if (kernel_command_line().is_smp_enabled() && arch_is_smp_supported()) {
    arch_enable_application_processors();
}
```

like, init_ap and init_finished could be moved to Arch/x86/SMPInit.cpp, along with this check here.

Reply:
I agree we should have some kind of abstraction for this, instead of adding ifdefs to the init.cpp. I added a FIXME for that.